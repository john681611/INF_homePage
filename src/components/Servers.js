import React from 'react'

const Servers = (props) => {
  return (
    <section id="servers" class="section">
      <div class="section__inner">
        <h2>Servers</h2>
        {props.servers.map(server =>
          <span>
            <h3>
              {server.name}
            </h3>
            <p>
              <a href={server.info.link} target="_blank" rel="noopener noreferrer nofollow" title={server.info.link} class="postlink">
                <img alt="" resizemod="off" class="lazyload" data-src={server.info.src} border="0" width="100%" title="" />
              </a>
            </p>
            {server.file ?
              <span>
                <a class="btn" href={`https://ironfists.azurewebsites.net/modlist/${server.id}`} download={`INF_${server.name}_preset.html`}>Download Preset</a>
                <p class="smalltxt">Drag and drop file into Launcher</p>
              </span>
              : null}
            {Object.keys(server.mods).map(key =>
              <span>
                {server.mods[key].length > 0 ?
                  <span>
                    <h4>
                      {key}
                    </h4>
                    <p>
                      {server.mods[key].map(mod =>
                        <a class="btn" href={`https://steamcommunity.com/sharedfiles/filedetails/?id=${mod.steamID}`} target="_blank" rel="noopener noreferrer">
                          {mod.name}
                        </a>
                      )}
                    </p>
                  </span>
                  : null}
              </span>
            )}
          </span>
        )}
      </div>
    </section>
  )
}

export default Servers