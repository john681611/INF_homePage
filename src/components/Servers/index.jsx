import React from 'react'

const Servers = (props) => {
  return (
    <section id="servers" className="section">
      <div className="section__inner">
        <h2>Servers</h2>
        {props.servers.map(server =>
          <span key={server.name}>
            <h3>
              {server.name}
            </h3>
            <p>
              <a href={server.info.link} target="_blank" rel="noopener noreferrer nofollow" title={server.info.link} className="postlink">
                <img alt="" resizemod="off" className="lazyload" data-src={server.info.src} border="0" width="100%" title="" />
              </a>
            </p>
            {server.file ?
              <span>
                <a className="btn" href={`https://ironfists.azurewebsites.net/modlist/${server.id}`} download={`INF_${server.name}_preset.html`}>Download Preset</a>
                <p className="smalltxt">Drag and drop file into Launcher</p>
              </span>
              : null}
            {Object.keys(server.mods).map(key =>
              <span  key={key}>
                {server.mods[key].length > 0 ?
                  <span>
                    <h4>
                      {key}
                    </h4>
                    <p>
                      {server.mods[key].map(mod =>
                        <a  key={mod.steamID} className="btn mod" href={`https://steamcommunity.com/sharedfiles/filedetails/?id=${mod.steamID}`} target="_blank" rel="noopener noreferrer">
                          {mod.name}
                        </a>
                      )}
                    </p>
                  </span>
                  : null}
              </span>
            )}
          </span>
        )}
      </div>
    </section>
  )
}

export default Servers