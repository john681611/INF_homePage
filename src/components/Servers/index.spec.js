import React from 'react';
import { shallow } from 'enzyme';
import Servers from './';

describe('Servers', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Servers
      servers={[]}
      />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render 1 section', () => {
    expect(wrapper.find('section').length).toEqual(1);
  });

  it('should render each server name and url', () => {
    wrapper.setProps({ servers: [{
        name: 'Alpha',
        info: {link: 'bla', src: 'meh'},
        mods:{},
        file: true,
        id: '1'
    },
    {
        name: 'Bravo',
        info: {link: 'pah', src: 'beh'},
        mods: {}
    }
    ]});
    const name = wrapper.find('h3');
    expect(name.length).toEqual(2);
    expect(name.at(0).text()).toEqual('Alpha');
    expect(name.at(1).text()).toEqual('Bravo');

    const link = wrapper.find('.postlink');
    expect(link.length).toEqual(2);
    expect(link.at(0).prop('href')).toEqual('bla');
    expect(link.at(1).prop('href')).toEqual('pah');

    const image = wrapper.find('.lazyload');
    expect(image.length).toEqual(2);
    expect(image.at(0).prop('data-src')).toEqual('meh');
    expect(image.at(1).prop('data-src')).toEqual('beh');

    const file = wrapper.find('.btn');
    expect(file.length).toEqual(1);
    expect(file.at(0).prop('href')).toEqual('https://ironfists.azurewebsites.net/modlist/1');
    expect(file.at(0).prop('download')).toEqual('INF_Alpha_preset.html')
  });

  it('should render mandatory and optional mods for each group', () => {
    wrapper.setProps({ servers: [{
        name: 'Alpha',
        info: {link: 'bla', src: 'meh'},
        mods:{
            required: [{steamID: 1234, name:'Bob'}],
            optional: [{steamID: 4567, name:'sid'}]
        }
    }]});
    const name = wrapper.find('h4');
    expect(name.length).toEqual(2);
    expect(name.at(0).text()).toEqual('required');
    expect(name.at(1).text()).toEqual('optional');

    const modeName = wrapper.find('.mod');
    expect(modeName.length).toEqual(2);
    expect(modeName.at(0).text()).toEqual('Bob');
    expect(modeName.at(0).prop('href')).toEqual('https://steamcommunity.com/sharedfiles/filedetails/?id=1234');
    expect(modeName.at(1).text()).toEqual('sid');
    expect(modeName.at(1).prop('href')).toEqual('https://steamcommunity.com/sharedfiles/filedetails/?id=4567');

  });
});