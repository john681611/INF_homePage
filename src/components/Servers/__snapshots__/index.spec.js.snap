// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Servers should render correctly 1`] = `ShallowWrapper {}`;
